### PR TITLE
[Cadence Schedules] Adding fuzz tests for the schedule mappers

### PR DESCRIPTION
**Pull request by abhishekj720**
**What changed?**
Added fuzz tests to test out round trip schedule tests by running random numbers.

**Why?**
For dynamic ranges of value to test to not skip any edge case

**How did you test it?**
Tests

**Potential risks**
N/A: only tests

**Release notes**
N/A: only tests

**Documentation Changes**
N/A
---

## Reviewer Validation

**PR Description Quality** (check these before reviewing code):

- [ ] **"What changed"** provides a clear 1-2 line summary
  - [ ] Project Issue is linked
- [ ] **"Why"** explains the full motivation with sufficient context
- [ ] **Testing is documented:**
  - [ ] Unit test commands are included (with exact `go test` invocation)
  - [ ] Integration test setup/commands included (if integration tests were run)
  - [ ] Canary testing details included (if canary was mentioned)
- [ ] **Potential risks** section is thoughtfully filled out (or legitimately N/A)
- [ ] **Release notes** included if this completes a user-facing feature
- [ ] **Documentation** needs are addressed (or noted if uncertain)

### CODE REVIEW

File: common/types/mapper/proto/schedule_test.go [hunk 1]
```diff
@@ -52,3 +55,103 @@ func TestSchedulePolicies(t *testing.T) {
...
+		fuzzer := scheduleFuzzer(f)
+		var orig *types.StartWorkflowAction
+		fuzzer.Fuzz(&orig)
+		out := ToStartWorkflowAction(FromStartWorkflowAction(orig))
+		assert.Equal(t, orig, out, "StartWorkflowAction did not survive round-tripping")
+
+		if orig == nil {
+			return "nil"
+		}
+		if orig.WorkflowType == nil && orig.TaskList == nil && orig.Input == nil && orig.WorkflowIDPrefix == "" {
```
[the remark this answers is not in the document]
Comment (abhishekj720) [reply common/types/mapper/proto/schedule_test.go:116]: The suggested fix would make the "empty" bucket unreachable: StartWorkflowAction has 9 fields, so with NilChance(0.3) the probability of all being zero simultaneously is ~0.3^8 ≈ 0.00007 per iteration, meaning "empty" would never be hit in 10k tries. 
This matches the existing pattern in TestRegisterDomainRequestFuzz which also checks only a representative subset of fields (3 out of ~15). The round-trip assertion runs regardless of classification, so correctness is not affected.
